wordpress-mobile/WordPress-MediaPicker-Android
# Hack week: Update target SKD to 33

This PR updates the target SDK to version 33 and implements the new granular media access permissions (images, video, audio), described in the [documentation](https://developer.android.com/about/versions/13/behavior-changes-13).

On devices running older Android versions (< API 33), the original single `Storage` permission is requested, which gives access to all media. On versions 33 and up, new granular permissions are requested (images and video are requested in a single permission request, audio in another one).

| Old version | New version |
| --- | --- |
| ![image](https://user-images.githubusercontent.com/1522856/228355430-48756346-fc9e-4efc-8cda-f26665df0603.png) | ![image](https://user-images.githubusercontent.com/1522856/228355568-2ba9f8e5-cebc-4e1a-9f28-3497bf616b3b.png) |

**To test:**

To test this PR, please prepare 2 devices/emulators with Android 33 and <33. Also make sure the devices have at least one image to display something (you can take a picture in the Camera app).

_API 33:_
1. Start with a clean installation of the sample app
2. Tap on the Local Device Picker button
3. Tap on the Allow button
4. Notice a permission request for `music and audio` appears
5. Grant the access
6. Notice another permission request appears, now for `photos and videos`
7. Grant the access
8. Notice the images stored on a device are listed
9. If you clear the app storage and try this again but deny the permissions (twice), you'll see the screen show on the screenshot above

_API <33:_
1. Repeat the same steps as for API 33
2. When you tap on the Allow button, you'll get a single permission request mentioning `photos, media and files`
3. If you deny the permission, you'll see the screen above
